tests/integration/local/gateway_local_test.py

```diff
@@ -253,3 +339,45 @@ def test_enclave_connect_to_gateway(faker: Faker, gateway, enclave):
     assert (
         proxy_enclave_client.api.endpoints.keys() == enclave_client.api.endpoints.keys()
     )
+
+
+@pytest.mark.local_node
+@pytest.mark.parametrize(
+    "gateway_association_request_auto_approval", [False], indirect=True
+)
+def test_repeated_association_requests_peers_health_check(
+    gateway_association_request_auto_approval, domain
+):
+    _, gateway = gateway_association_request_auto_approval
+    gateway_client: GatewayClient = gateway.login(
+        email="info@openmined.org",
+        password="changethis",
+    )
+    domain_client: DomainClient = domain.login(
+        email="info@openmined.org",
+        password="changethis",
+    )
+
+    result = domain_client.connect_to_gateway(handle=gateway)
+    assert isinstance(result, Request)
+
+    result = domain_client.connect_to_gateway(handle=gateway)
+    assert isinstance(result, Request)
+
+    r = gateway_client.api.services.request.get_all()[-1].approve()
+    assert isinstance(r, SyftSuccess)
+
+    result = domain_client.connect_to_gateway(handle=gateway)
+    assert isinstance(result, SyftSuccess)
+
+    # the gateway client checks that the peer is associated
+    res = gateway_client.api.services.network.check_peer_association(
+        peer_id=domain_client.id
+    )
+    assert isinstance(res, NodePeerAssociationStatus)
+    assert res.value == "PEER_ASSOCIATED"
+
+    # check for peer connection status
+    time.sleep(PeerHealthCheckTask.repeat_time + 1)
```

Comment:
need to wait for sometime to make the peer connection status check passes. However, this slow down the test

Reply:
we should wait for at least `PeerHealthCheckTask.repeat_time * 2`